mirascope/core/groq/tool.py

```diff
@@ -74,6 +74,7 @@ def from_tool_call(cls, tool_call: ChatCompletionMessageToolCall) -> GroqTool:
         Args:
             tool_call: The Groq tool call from which to construct this tool instance.
         """
-        model_json = jiter.from_json(tool_call.function.arguments.encode())
-        model_json["tool_call"] = tool_call.model_dump()
+        model_json = {"tool_call": tool_call}
```

Comment:
this was previously `tool_call.model_dump()` so we should make sure to keep that

that's my mistake in my previous suggestion

Reply:
I'll let you tidy this one up. Thanks @willbakst